react/src/hooks/useMemoWithPrevious.tsx

```diff
@@ -0,0 +1,36 @@
+import { DependencyList, useEffect, useMemo, useRef, useState } from 'react';
+
+export const useMemoWithPrevious = <T,>(
+  factory: () => T,
+  { deps, initialPrev }: { deps: DependencyList; initialPrev?: T },
+) => {
```

Comment:
I'd suggest to design as below for consistency.
```suggestion
export const useMemoWithPrevious = <T,>(
  factory: () => T,
  deps: DependencyList,
  { initialPrev }: { initialPrev?: T },
) => {
```